proof/crefine/AARCH64/VSpace_C.thy

```diff
@@ -2871,8 +2848,7 @@ lemma vgicUpdateLR_ccorres:
   apply (rule ccorres_grab_asm)
   apply (simp add: vgicUpdateLR_def vgicUpdate_def)
   apply vcpuUpdate_ccorres
-  supply from_bool_eq_if[simp] from_bool_eq_if'[simp] from_bool_0[simp]
-  apply (fastforce simp: virq_to_H_def cvcpu_vppi_masked_relation_def split: if_split)
+  apply (auto simp: max_armKSGICVCPUNumListRegs_val split: if_split)
```

Comment:
I'm never quite comfortable with these. I know why they're needed, but a part of me wants to yell "abstraction violation" ... not clear how we can go about cleaning this up except via static include functions or enums in C